lib/StackAnalysis/StackAnalysis.cpp

```diff
@@ -324,6 +354,655 @@ void commitToModel(GeneratedCodeBasicInfo &GCBI,
+  auto *Call = getFunctionCall(Term);
+
+  // If the function callee is null, we are dealing with an indirect call.
+  auto *Next = getFunctionCallCallee(Term);
+
+  // What is the function type of the callee?
+  FunctionTypeValue Ty = Oracle.getFunctionType(Next);
+
+  switch (Ty) {
+  case FunctionTypeValue::Regular: {
+    BranchInst *Br = BranchInst::Create(getFallthrough(Term));
+    ReplaceInstWithInst(Term, Br);
+
+    // Extract MetaAddress of `newpc` of the call-site and the callee.
+    auto [NewPCCallSite, _] = getPC(Br);
+    auto NewPCCallee = Next ? getBasicBlockPC(Next) : MetaAddress::invalid();
+
+    // What are the registers clobbered by the callee?
+    const auto &ClobberedRegs = Oracle.getRegistersClobbered(Next);
+    IRBuilder<> Builder(Br);
+
+    // Mark end of basic block with a pre-hook call.
+    StructType *MetaAddressTy = MetaAddress::getStruct(&M);
+    SmallVector<Value *, 2> Args = { NewPCCallSite.toConstant(MetaAddressTy),
+                                     NewPCCallee.toConstant(MetaAddressTy) };
+    auto *Last = Builder.CreateCall(PreHookMarker, Args);
+
+    // Prevent the store instructions from being optimized out by storing
+    // the rval of a call to an opaque function into the clobbered registers.
+    OFPRegistersClobbered.addFnAttribute(Attribute::ReadOnly);
+    OFPRegistersClobbered.addFnAttribute(Attribute::NoUnwind);
+
+    for (GlobalVariable *Reg : ClobberedRegs) {
+      auto *CSVTy = Reg->getType()->getPointerElementType();
+      auto *RegsClobbOpqF = OFPRegistersClobbered
+                              .get(BB->getParent()->getName(),
+                                   CSVTy,
+                                   {},
+                                   "registers_clobbered");
+      Builder.CreateStore(Builder.CreateCall(RegsClobbOpqF), Reg);
+    }
+
+    // Adjust back the stack pointer.
+    switch (GCBI->arch()) {
+    case llvm::Triple::x86:
+    case llvm::Triple::x86_64: {
+      auto *SPCSV = M.getGlobalVariable(GCBI->spReg()->getName(), true);
+      auto *SP = Builder.CreateLoad(SPCSV);
+
+      // What is the elected final stack offset?
+      const auto &FSO = Oracle.getElectedFSO(Next);
+      Value *Offset = GCBI->arch() == llvm::Triple::x86 ?
+                        Builder.getInt32(*FSO) :
+                        Builder.getInt64(*FSO);
+      auto *Inc = Builder.CreateAdd(SP, Offset);
+      Builder.CreateStore(Inc, SPCSV);
+      break;
+    }
+    default: {
+      break;
+    }
+    }
+
+    // Mark end of basic block with a post-hook call.
+    Builder.CreateCall(PostHookMarker, Args);
+
+    Split = BB->splitBasicBlock(Last->getPrevNode(),
+                                BB->getName() + Twine("__summary"));
+    break;
+  }
+
+  case FunctionTypeValue::NoReturn: {
+    CallInst *Abort = CallInst::Create(M.getFunction("abort"));
+    new UnreachableInst(BB->getContext(), BB);
+    ReplaceInstWithInst(Term, Abort);
+    break;
+  }
```

Comment:
This is not correct.
You still need to do everything you do for a regular function, except that instead of jumping to the fallthrough you need to abort.